src/Bicep.LangServer/Handlers/BicepCodeActionHandler.cs

```diff
@@ -67,12 +74,88 @@ public override Task<CommandOrCodeActionContainer> Handle(CodeActionParams reque
 
             commandOrCodeActions.AddRange(analyzerDiagnostics);
 
+            var diagnosticsOfTypeWarningAndInfo = diagnostics
+                .Where(diagnostic =>
+                    (diagnostic.Level == DiagnosticLevel.Warning || diagnostic.Level == DiagnosticLevel.Info) &&
+                    (diagnostic.Span.ContainsInclusive(requestStartOffset) ||
+                    diagnostic.Span.ContainsInclusive(requestEndOffset) ||
+                    (requestStartOffset <= diagnostic.Span.Position && diagnostic.GetEndPosition() <= requestEndOffset)))
+                .OfType<IDiagnostic>();
+
+            HashSet<string> diagnosticCodesToSuppressInline = new();
+
+            foreach (IDiagnostic diagnostic in diagnosticsOfTypeWarningAndInfo)
+            {
+                if (!diagnosticCodesToSuppressInline.Contains(diagnostic.Code))
+                {
+                    diagnosticCodesToSuppressInline.Add(diagnostic.Code);
+
+                    var commandOrCodeAction = DisableDiagnostic(documentUri, diagnostic.Code, semanticModel.SourceFile, diagnostic.Span, compilationContext.LineStarts);
+
+                    if (commandOrCodeAction is not null)
+                    {
+                        commandOrCodeActions.Add(commandOrCodeAction);
+                    }
+                }
+            }
+
             return Task.FromResult(new CommandOrCodeActionContainer(commandOrCodeActions));
         }
 
+        private static CommandOrCodeAction? DisableDiagnostic(DocumentUri documentUri,
+            DiagnosticCode diagnosticCode,
+            BicepFile bicepFile,
+            TextSpan span,
+            ImmutableArray<int> lineStarts)
+        {
+            if (diagnosticCode.String is null)
+            {
+                return null;
+            }
+
+            var disableNextLineDiagnosticDirectivesCache = bicepFile.DisableNextLineDiagnosticDirectivesCache;
+            (int diagnosticLine, _) = TextCoordinateConverter.GetPosition(bicepFile.LineStarts, span.Position);
+
+            TextEdit? textEdit;
+            int previousLine = diagnosticLine - 1;
```

Comment:
What happens if `diagnosticLine` is 0?

Reply:
We would insert #disable-next-line ..  at line 0, by executing else block. The integration test I added - VerifyDisableNextLineDiagnosticsCodeActionInvocation is an example. Let me know if you have any concerns.